### PR TITLE
Basic CI Setup

### LINKED ISSUE #2: Setup CI
We need a github action to build and run the tests.

### PR DESCRIPTION
**Pull request by VibhavariMandal-sumo**
This is a first attempt at setting up a build system for building opamp-java-library. The workflow triggers when a pull request is raised or a code is pushed.

This checks out the proto submodules, builds the proto repo & includes all the proto classes in the final jar of opamp-java-library. 

After building the library, it is published to a maven local repository. For publishing to a remote maven repo, I suggest we do it in a separate PR, once we finalise where we need to publish.

$ tree /tmp/opamp-java/io/opentelemetry/opamp-java/1.0.0/
```
/tmp/opamp-java/io/opentelemetry/opamp-java/1.0.0/

├── opamp-java-1.0.0-javadoc.jar
├── opamp-java-1.0.0-javadoc.jar.md5
├── opamp-java-1.0.0-javadoc.jar.sha1
├── opamp-java-1.0.0-javadoc.jar.sha256
├── opamp-java-1.0.0-javadoc.jar.sha512
├── opamp-java-1.0.0-sources.jar
├── opamp-java-1.0.0-sources.jar.md5
├── opamp-java-1.0.0-sources.jar.sha1
├── opamp-java-1.0.0-sources.jar.sha256
├── opamp-java-1.0.0-sources.jar.sha512
├── opamp-java-1.0.0.jar
├── opamp-java-1.0.0.jar.md5
├── opamp-java-1.0.0.jar.sha1
├── opamp-java-1.0.0.jar.sha256
├── opamp-java-1.0.0.jar.sha512
├── opamp-java-1.0.0.module
├── opamp-java-1.0.0.module.md5
├── opamp-java-1.0.0.module.sha1
├── opamp-java-1.0.0.module.sha256
├── opamp-java-1.0.0.module.sha512
├── opamp-java-1.0.0.pom
├── opamp-java-1.0.0.pom.md5
├── opamp-java-1.0.0.pom.sha1
├── opamp-java-1.0.0.pom.sha256
└── opamp-java-1.0.0.pom.sha512
```

Resolves https://github.com/open-telemetry/opamp-java/issues/2

### CODE REVIEW

**Comment by VibhavariMandal-sumo** [comment]
cc @tigrannajaryan @jlegoff

**Comment by jlegoff** [comment]
@jack-berg could you review it?

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  push:
+    branches:
+      - main
+      - release/*
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (jlegoff) [inline .github/workflows/build-and-publish.yml]: I would use a specific version to make sure the build is reproducible. I see other OTEL projects use ubuntu-20.04, e.g. [here](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/.github/workflows/main-build.yml)

File: build.gradle.kts [hunk 2]
```diff
@@ -1,8 +1,33 @@
 plugins {
   id("otel.java-conventions")
+  id("java-library")
+  id("maven-publish")
```
Comment (jlegoff) [inline build.gradle.kts]: I think we should use the otel plugin, like [here](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/build.gradle.kts#L13)

File: build.gradle.kts [hunk 3]
```diff
@@ -1,8 +1,33 @@
...
   mavenCentral()
   mavenLocal()
 }
+
+dependencies {
+  implementation(project(":proto"))
+}
+
+group = "io.opentelemetry"
+version = "1.0.0"
```
Comment (jlegoff) [inline build.gradle.kts]: we shouldn't use 1.0.0 here, but rather the release version. I'm not sure what's the best way to achieve this, perhaps we could add do something like [this](https://github.com/open-telemetry/opentelemetry-java/blob/main/version.gradle.kts)

**Review by jlegoff — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (jack-berg) [reply build.gradle.kts]: The otel plugin isn't published. The source code lives in the repository in `buildsrc` [here](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts).

Still, we should similarly organize the project with convention plugins.

**Comment by jack-berg** [comment]
The `opentelemetry-java` project publishes snapshots on each merge to master, but publication of non-snapshot artifacts is manually controlled. I think we should mimic that here.

**Comment by VibhavariMandal-sumo** [comment]
> The `opentelemetry-java` project publishes snapshots on each merge to master, but publication of non-snapshot artifacts is manually controlled. I think we should mimic that here.

1) @jack-berg , do you suggest publishing to a remote maven repo, if yes, can you guide me how do I obtain url and credential for that? I read the credentials are present in https://opentelemetry.1password.com/signin, but I dont have access to this.

File: build.gradle.kts (hunk 2 above)
Comment (VibhavariMandal-sumo) [reply build.gradle.kts:4]: @jack-berg are you suggesting writing similar plugin as [this](https://github.com/open-telemetry/opentelemetry-proto-java/blob/main/buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts) in this repo as well?
If yes, can you please tell me advantage of using it over standard maven publish pugin?

**Comment by VibhavariMandal-sumo** [comment]
I will raise a new PR only for PR builder workflow

**Pull request closed without merging**